tests/core/management/commands/test_send_push_notifications.py

```diff
@@ -1,33 +1,155 @@
 from pytest_django.fixtures import SettingsWrapper
+from requests_mock.mocker import Mocker
+
+from integreat_cms.cms.models import (
+    Language,
+    LanguageTreeNode,
+    PushNotification,
+    PushNotificationTranslation,
+    Region,
+)
+from integreat_cms.firebase_api.firebase_api_client import FirebaseApiClient
 
 from ..utils import get_command_output
 
 
-@pytest.mark.django_db
-def test_push_notifications_disabled(settings: SettingsWrapper) -> None:
+class TestSendPushNotification:
     """
-    Ensure that disabled push notifications cause an error
+    Test that cover the send_push_notification management command and its filter mechanisms
     """
-    settings.FCM_ENABLED = False
-    with pytest.raises(CommandError) as exc_info:
-        assert not any(get_command_output("send_push_notifications"))
-    assert str(exc_info.value) == "Push notifications are disabled"
 
+    @pytest.mark.django_db
+    def test_push_notifications_disabled(self, settings: SettingsWrapper) -> None:
+        """
+        Ensure that disabled push notifications cause an error
+        """
+        settings.FCM_ENABLED = False
+        with pytest.raises(CommandError) as exc_info:
+            assert not any(get_command_output("send_push_notifications"))
+        assert str(exc_info.value) == "Push notifications are disabled"
 
-@pytest.mark.django_db
-def test_push_notifications_nonexisting_testregion(settings: SettingsWrapper) -> None:
-    """
-    Ensure that an error is caused when the system runs in debug mode but the test region does not exist
-    """
-    settings.TEST_REGION_SLUG = "non-existing"
-    settings.DEBUG = True
-    with pytest.raises(CommandError) as exc_info:
-        assert not any(get_command_output("send_push_notifications"))
-    assert (
-        str(exc_info.value)
-        == f"The system runs with DEBUG=True but the region with TEST_REGION_SLUG={settings.TEST_REGION_SLUG} does not exist."
-    )
+    @pytest.mark.django_db
+    def test_push_notifications_nonexisting_testregion(
+        self,
+        settings: SettingsWrapper,
+    ) -> None:
+        """
+        Ensure that an error is caused when the system runs in debug mode but the test region does not exist
+        """
+        settings.TEST_REGION_SLUG = "non-existing"
+        settings.DEBUG = True
+        with pytest.raises(CommandError) as exc_info:
+            assert not any(get_command_output("send_push_notifications"))
+        assert (
+            str(exc_info.value)
+            == f"The system runs with DEBUG=True but the region with TEST_REGION_SLUG={settings.TEST_REGION_SLUG} does not exist."
+        )
+
+    patch: Any = None
+
+    def setup_method(self) -> None:
+        self.patch = patch.object(
+            FirebaseApiClient, "_get_access_token", return_value="secret access token"
+        )
+        self.patch.start()
+
+    def teardown_method(self) -> None:
+        self.patch.stop()
+        self.patch = None
```

Comment:
I think it is more readable if the setup related stuff is at the top

Reply:
Absolutely :)